src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -188,6 +188,20 @@ case object ThurloeDatabaseConnector extends DataAccess {
     val setup = DBIO.seq(keyValuePairTable.schema.create)
     Await.result(database.run(setup), Duration.Inf)
   }
+
+  def status(): Future[Unit] = {
+    // Check database connection by selecting version; if it's successful return successful future
+    // Otherwise, return a failed future
+    val action = sql"select version ()".as[String]
+    val versionFuture: Future[Vector[String]] = database.run(action.transactionally)
+
+    versionFuture map { _ =>
+      ()
+    } recoverWith {
+      case _ =>
+        Future.failed(DatabaseConnectionException())
+    }
+  }
```

Comment:
you don't actually use DatabaseConnectionException anywhere and masking the underlying exception is probably not a great idea.

you can condense this function:
  def status(): Future[Unit] = {
    // Check database connection by selecting version; if it's successful return successful future
    // Otherwise, return a failed future
    val action = sql"select version ()".as[String]
    database.run(action.transactionally)  map { _ => Unit }
  }